mat/band.go

```diff
@@ -136,6 +136,12 @@ func NewBandDense(r, c, kl, ku int, data []float64) *BandDense {
 	}
 }
 
+// NewDiagonal is a convenience function that returns a diagonal matrix represented by a
+// BandDense. The length of data must be min(r, c) otherwise NewDiagonal will panic.
+func NewDiagonal(r, c int, data []float64) *BandDense {
```

Comment:
An alternative is to have `NewDiagonal(r, c int, data []float64) Matrix` and return either a `*BandDense` or a `*SymBandDense` depending on whether r == c.